File: represent/bench/c_flat/SpecGenBench_0320__SpecGenBench__common__IsSuffix__IsSuffix.c
#include <string.h>

int isSuffix(const char * pat, const char * txt);

int isSuffix(const char * pat, const char * txt) {

        int i = ((int)strlen(pat)) - 1;

        while(i >= 0) {
	    int j = i - ((int)strlen(pat)) + ((int)strlen(txt));
            if(j < 0 || pat[i) != txt[j))
                return 0;
            i = i - 1;
        }

        return 1;
}

File: represent/bench/java_flat/SpecGenBench_0320__SpecGenBench__common__IsSuffix__IsSuffix.java
public class IsSuffix {

    public boolean isSuffix (String pat, String txt) {

        int i = pat.length() - 1;

        while(i >= 0) {
	    int j = i - pat.length() + txt.length();
            if(j < 0 || pat.charAt(i) != txt.charAt(j))
                return false;
            i = i - 1;
        }

        return true;
    }

}
